src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Escaping.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text.Encodings.Web;
+
+namespace System.Text.Json
+{
+    internal static partial class JsonHelpers
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static byte[] GetEscapedPropertyNameSection(ReadOnlySpan<byte> utf8Value, JavaScriptEncoder? encoder)
```

Comment:
Since dictionary keys are specified in the JSON and not known ahead of time, I'm not sure if caching would be beneficial (more memory would be required, and the cache would need to be capped).